droidlet/perception/robot/tests/test_perception.py

```diff
@@ -134,19 +134,17 @@ def test_handler_dedupe(self):
         self.assertGreaterEqual(len(detections), 5)  # 9 exactly
         # insert once to setup dedupe tests
         self.deduplicator(detections, [])
-        for obj in detections:
-            obj.save_to_memory(self.agent.memory)
+        self.agent.memory.update(namedtuple("detections", ["new_detections"])(detections))
 
         objs_init = DetectedObjectNode.get_all(self.agent.memory)
 
         # Insert with dedupe
         previous_objects = DetectedObjectNode.get_all(self.agent.memory)
+        detectionTuple = namedtuple("detections", ["new_detections", "updated_detections"])
         if previous_objects is not None:
             new_objects, updated_objects = self.deduplicator(detections, previous_objects)
-            for obj in new_objects:
-                obj.save_to_memory(self.agent.memory)
-            for obj in updated_objects:
-                obj.save_to_memory(self.agent.memory, update=True)
+            detection_output = detectionTuple(new_objects, updated_objects)
```

Comment:
this would become

```python
output = PerceptionOutput(new_objects=new_objects, updated_objects=updated_objects)
self.agent.memory.update(output)
```